synapse/cortex.py

```diff
@@ -1027,6 +1028,45 @@ async def initServiceStorage(self):
 
         self._initVaults()
 
+    def hasModelSave(self, iden):
+        return self.slab.has(s_common.uhex(iden), db='model:saves')
+
+    def getModelSave(self, iden):
+        byts = self.slab.get(s_common.uhex(iden), db='model:saves')
+        if byts is not None:
+            return s_msgpack.un(byts)
+
+    async def addModelSave(self, iden, model):
+        return await self._push('model:save:add', iden, model)
+
+    @s_nexus.Pusher.onPush('model:save:add')
+    async def _addModelSave(self, iden, model):
+        lkey = s_common.uhex(iden)
+        envl = {
+            'fmt': 1,
+            'iden': iden,
+            'model': model,
+            'synapse': s_version.verstring,
```

Comment:
I think we need to send this from the top half to avoid a mirror saving with a newer synapse version.

Reply:
All of the extended model stuff probably needs to be methods that run on the leader, I think there are a bunch of potential issues